qemu/tests/balloon_service.py

```diff
@@ -131,57 +57,71 @@ def balloon_memory(vm, device_path, mem_check):
+                    balloon_test.memory_status_check('stat-free-memory')
                 if quit_after_test:
                     return
 
             repeat_times -= 1
 
-    timeout = int(params.get("login_timeout", 360))
     mem_check = params.get("mem_check", "yes")
 
     error_context.context("Boot guest with balloon device", logging.info)
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
 
-    sleep_time = int(params.get("polling_sleep_time", 20))
-    base_path = params.get("base_path", "/machine/peripheral/")
-    device = params.get("balloon", "balloon0")
-    device_path = os.path.join(base_path, device)
-    set_balloon_property = params.get("set_balloon_property",
-                                      "guest-stats-polling-interval")
-    get_balloon_property = params.get("get_balloon_property", "guest-stats")
-    polling_interval = int(params.get("polling_interval", 2))
+    session = vm.wait_for_login()
 
-    session = vm.wait_for_login(timeout=timeout)
     if params['os_type'] == 'windows':
         driver_name = params.get("driver_name", "balloon")
-        utils_test.qemu.setup_win_driver_verifier(driver_name, vm, timeout)
-
+        utils_test.qemu.windrv_verify_running(session, test, driver_name)
+        utils_test.qemu.setup_win_driver_verifier(driver_name, vm)
+        balloon_test = BallooningTestWin(test, params, env)
         error_context.context("Config balloon service in guest",
                               logging.info)
-        drive_letter = get_disk_vol(session)
-        config_balloon_service(session, drive_letter)
+        balloon_test.configure_balloon_service(session)
+    else:
+        balloon_test = BallooningTestLinux(test, params, env)
 
     try:
-        error_context.context("Enable polling in qemu", logging.info)
-        vm.monitor.qom_set(device_path, set_balloon_property, polling_interval)
-        time.sleep(sleep_time)
-        get_polling_output = vm.monitor.qom_get(device_path,
-                                                get_balloon_property)
-        memory_check(vm, get_polling_output, 'stat-total-memory')
+        error_context.context("Check memory status via balloon service", logging.info)
+        balloon_test.memory_status_check('stat-total-memory')
 
         error_context.context("Balloon vm memory in loop", logging.info)
-        balloon_memory(vm, device_path, mem_check)
+        balloon_memory(vm, mem_check)
+
+        blnsrv_operation = params.objects("blnsrv_operation")
+        if blnsrv_operation:
+            error_context.context("%s balloon service" % blnsrv_operation[0], logging.info)
+            balloon_test.operate_balloon_service(session, blnsrv_operation[0])
+
+            output = balloon_test.operate_balloon_service(session, "status")
+            if ((params.get("check_status_after_oper", "no") == "yes") and not
+                re.search(params.get("expected_status_after_oper"),
+                          output.lower(), re.M)):
```

Comment:
lijin, this format should be better. 
if ((params.get("check_status_after_oper", "no") == "yes") and
         not re.search(params.get("expected_status_after_oper"), output.lower(), re.M)):

Reply:
This is will this line too long.